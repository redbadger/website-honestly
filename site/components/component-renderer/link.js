// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
