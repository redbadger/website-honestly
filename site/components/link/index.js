// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
