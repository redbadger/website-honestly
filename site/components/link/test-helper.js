// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
