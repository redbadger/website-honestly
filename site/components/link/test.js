// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
