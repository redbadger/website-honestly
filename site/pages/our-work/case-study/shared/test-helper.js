// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
