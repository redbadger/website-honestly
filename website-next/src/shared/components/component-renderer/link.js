import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
