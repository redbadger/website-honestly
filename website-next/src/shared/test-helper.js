import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
