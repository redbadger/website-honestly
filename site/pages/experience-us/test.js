import React from 'react';
import { render } from 'enzyme';

import { Context } from '../../components/link/test-helper';

import ExperienceUsPage from './index';

const goldCoinPages = [
  {
    consultants: ['mr-sitemap', 'milo-castro', 'clementine-brown'],
    duration: '1 hour',
    headerAlt: null,
    headerImage:
      'https://images.prismic.io/rb-website-stage/b378bba2-8c68-4734-b657-455e3af357f1_Headerimage.png?auto=compress,format&rect=159,0,2560,948&w=2560&h=948',
    hubspotForm: {
      portalId: 4210858,
      guid: 'a2c3b495-7093-47af-a6b6-bd4d82b9fe94',
      name: 'Gold coins: service mesh',
      cssClass: 'hs-form stacked',
      submitText: 'Book now',
    },
    location: 'Your office, or ours',
    price: 'Free',
    slug: 'service-mesh',
    subTitle: 'Service Mesh',
    title: 'Service Mesh',
    type: 'lunch',
    whatIsIt:
      '<p>Service Mesh is an emerging approach to orchestrating microservices at scale. When designed and deployed well, it can help you solve some of the major challenges posed by complex system architectures. These areas include service discovery, security, networking and observability.</p>',
    whatWillYouLearn:
      '<ol><li>What is Service Mesh</li><li>How organisations can benefit by reducing complexity and improving efficiency</li><li>Where to start with Service Mesh for your organisation</li></ol>',
    whoIsItFor: '<p>CTOs, engineers, architects and technical product leads   </p>',
    whoWillRun: '<p>Red Badger’s Technical Directors </p>',
  },
  {
    consultants: ['mr-sitemap', 'milo-castro', 'clementine-brown'],
    duration: '1 day',
    headerAlt: null,
    headerImage:
      'https://images.prismic.io/rb-website-stage/b378bba2-8c68-4734-b657-455e3af357f1_Headerimage.png?auto=compress,format&rect=159,0,2560,948&w=2560&h=948',
    hubspotForm: {
      portalId: 4210858,
      guid: 'a2c3b495-7093-47af-a6b6-bd4d82b9fe94',
      name: 'Gold coins: service mesh',
      cssClass: 'hs-form stacked',
      submitText: 'Book now',
    },
    location: 'Your office, or ours',
    price: 'Free',
    slug: 'service-mesh',
    subTitle: 'Service Mesh',
    title: 'Service Mesh',
    type: 'lunch',
    whatIsIt:
      '<p>Service Mesh is an emerging approach to orchestrating microservices at scale. When designed and deployed well, it can help you solve some of the major challenges posed by complex system architectures. These areas include service discovery, security, networking and observability.</p>',
    whatWillYouLearn:
      '<ol><li>What is Service Mesh</li><li>How organisations can benefit by reducing complexity and improving efficiency</li><li>Where to start with Service Mesh for your organisation</li></ol>',
    whoIsItFor: '<p>CTOs, engineers, architects and technical product leads   </p>',
    whoWillRun: '<p>Red Badger’s Technical Directors </p>',
  },
  {
    consultants: ['mr-sitemap', 'milo-castro', 'clementine-brown'],
    duration: '1 week',
    headerAlt: null,
    headerImage:
      'https://images.prismic.io/rb-website-stage/b378bba2-8c68-4734-b657-455e3af357f1_Headerimage.png?auto=compress,format&rect=159,0,2560,948&w=2560&h=948',
    hubspotForm: {
      portalId: 4210858,
      guid: 'a2c3b495-7093-47af-a6b6-bd4d82b9fe94',
      name: 'Gold coins: service mesh',
      cssClass: 'hs-form stacked',
      submitText: 'Book now',
    },
    location: 'Your office, or ours',
    price: 'Free',
    slug: 'service-mesh',
    subTitle: 'Service Mesh',
    title: 'Service Mesh',
    type: 'lunch',
    whatIsIt:
      '<p>Service Mesh is an emerging approach to orchestrating microservices at scale. When designed and deployed well, it can help you solve some of the major challenges posed by complex system architectures. These areas include service discovery, security, networking and observability.</p>',
    whatWillYouLearn:
      '<ol><li>What is Service Mesh</li><li>How organisations can benefit by reducing complexity and improving efficiency</li><li>Where to start with Service Mesh for your organisation</li></ol>',
    whoIsItFor: '<p>CTOs, engineers, architects and technical product leads   </p>',
    whoWillRun: '<p>Red Badger’s Technical Directors </p>',
  },
  {
    consultants: ['mr-sitemap', 'milo-castro', 'clementine-brown'],
    duration: '2 weeks',
    headerAlt: null,
    headerImage:
      'https://images.prismic.io/rb-website-stage/b378bba2-8c68-4734-b657-455e3af357f1_Headerimage.png?auto=compress,format&rect=159,0,2560,948&w=2560&h=948',
    hubspotForm: {
      portalId: 4210858,
      guid: 'a2c3b495-7093-47af-a6b6-bd4d82b9fe94',
      name: 'Gold coins: service mesh',
      cssClass: 'hs-form stacked',
      submitText: 'Book now',
    },
    location: 'Your office, or ours',
    price: 'Free',
    slug: 'service-mesh',
    subTitle: 'Service Mesh',
    title: 'Service Mesh',
    type: 'lunch',
    whatIsIt:
      '<p>Service Mesh is an emerging approach to orchestrating microservices at scale. When designed and deployed well, it can help you solve some of the major challenges posed by complex system architectures. These areas include service discovery, security, networking and observability.</p>',
    whatWillYouLearn:
      '<ol><li>What is Service Mesh</li><li>How organisations can benefit by reducing complexity and improving efficiency</li><li>Where to start with Service Mesh for your organisation</li></ol>',
    whoIsItFor: '<p>CTOs, engineers, architects and technical product leads   </p>',
    whoWillRun: '<p>Red Badger’s Technical Directors </p>',
  },
];

describe('site/pages/join-us', () => {
  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <ExperienceUsPage goldCoinPages={goldCoinPages} />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
