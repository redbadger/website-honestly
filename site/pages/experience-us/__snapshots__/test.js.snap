// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/experience-us Renders 1`] = `
<div>
  <Social
    altText="An illustration an award we won."
    description="Meet our tech and design experts to find out how we can deliver value, build capability, and change your culture to increase business efficiency."
    metaImage="Homepage_2.jpg"
    title="Experience Red Badger"
    url="https://red-badger.com/what-we-do/experience-us/"
  />
  <div
    className="background"
  >
    <div
      className="content"
    >
      <h3
        className="h3"
      >
        Experience Red Badger
      </h3>
      <h1
        className="h1"
      >
        How long have you got?
      </h1>
      <TimeframeSlice
        currentWidth="desktop"
        timeframes={
          Object {
            "1 day": Object {
              "copy": "Want to shake up the way you do things? We can bring fresh thinking and new approaches to your projects. In just one day.",
              "heading": "Let’s explore",
              "pages": Array [],
            },
            "1 hour": Object {
              "copy": "The best way to know if we’re right for you is to meet up. Here are some suggestions if you’ve got an hour to spare.",
              "heading": "Let’s meet",
              "pages": Array [],
            },
            "1 week": Object {
              "copy": "We love digging into tricky problems. Got a week? Discover how we create value by working together.",
              "heading": "Let’s dig",
              "pages": Array [],
            },
            "2 weeks": Object {
              "copy": "Given two weeks we can explore issues you’re facing in more depth, and even make things we can test. Here’s how to get going.",
              "heading": "Let’s do",
              "pages": Array [],
            },
          }
        }
      />
      <OtherWays
        goldCoinPages={
          Array [
            Object {
              "consultants": Array [
                Object {
                  "image": "https://www.example.com/test-3.jpg",
                  "name": "test-consultant",
                  "profileUrl": "https://www.example.com/test-consultant",
                  "role": "test-role",
                },
              ],
              "duration": "test-duration",
              "headerAlt": "test-alt",
              "headerImage": "https://www.example.com/test.jpg",
              "hubspotForm": Object {
                "cssClass": "test-css",
                "formFields": Array [
                  Object {
                    "enabled": true,
                    "fieldType": "lunch",
                    "hidden": false,
                    "label": "test-label",
                    "labelHidden": false,
                    "name": "test-field-name",
                    "required": true,
                  },
                ],
                "guid": "test-guid",
                "inlineMessage": "test-inline",
                "name": "test-name",
                "portalId": 1111111,
                "submitText": "test-submit",
              },
              "location": "test-location",
              "previews": Array [
                Object {
                  "alt": "test-alt-2",
                  "duration": "test-duration-2",
                  "image": "https://www.example.com/test-2.jpg",
                  "subTitle": "test-subtitle-2",
                  "title": "test-title-2",
                  "url": "https://www.example.com/",
                },
              ],
              "price": "test-price",
              "slug": "test",
              "subTitle": "test-subtitle",
              "title": "test-title",
              "type": "lunch",
              "whatIsIt": "<p>test what is it</p>",
              "whatWillYouLearn": "<p>test what will you learn</p>",
              "whoIsItFor": "<p>test who is it for</p>",
              "whoWillRun": "<p>test who will run</p>",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
